### DISCUSSION TITLE
Authorize channel only to authenticated users via sqlauth

### DISCUSSION THREAD

**Discussion by lol-powa** [inspircd/inspircd#2082] [Support]
Hi evryone,
French speaker here thanks for your patience.

I set up an IRC server with Inspircd, working fine.
I would like to take this opportunity to thank everyone involved in this project: You are doing a wonderful job.

My problem, for which I request your help:

There are two ways to connect to my Inspircd:
- Directly with a client, “anonymously”
- By authenticating via a Mysql database (Mysql module + Sqlauth)

I cannot reserve a channel for members only authenticated via the database.

I tried with the services_account module but I cannot achieve the expected result: Authorize a channel only to users who have authenticated via sqlauth.

Thanks for reading and thanks for your help.

**Comment by genius3000** [comment]
I don't think there's a simple and direct way to accomplish this. `m_sqlauth` is only intended to control access to the server, it doesn't set any metadata on the user nor allow connect class matching as a result.
What might work (and this is an untested thought) and assuming the "anonymous" connections are allowed by using `allowpattern` in `m_sqlauth`. Disallow nickname changes (`m_disable`) and set a channel ban on the allowed pattern, so only the authenticated users can enter.
Now, I don't recall the side effects to disallow nickname changes, I think disabling NICK with `m_disable` acts only after the connection is registered which is good, but please test this thought before rolling it out live. :)

**Reply by lol-powa** [reply]
Hello,
Thank you very much for your answer.

The metadata or flag on sqlauth authenticated users was my first thinking obviously as it is the best and easier solution.

And you are right I'll take a look on allowpattern which was my second idea :-) and should work.

Do you think I could find a better solution if I use Anope or Atheme ?

Have a good sunday.

**Reply by genius3000** [reply]
> Do you think I could find a better solution if I use Anope or Atheme ?

Simple answer, yes you could. It depends on your exact needs. With a services package you could make use of `m_services_account` and only allow registered users to create (`m_restrictchans`) or join a channel, etc. You can still use the external SQL database for users with Anope and I'd assume but unsure on the same for Atheme. You would lose some of your server connection control that your have with `m_sqlauth` though something similar could be done by requiring SASL authentication and then denying or limiting the non-SASL connections.

Join our IRC support channel if you can, feel free to ping me. This might be easier to discuss back and forth on IRC, figure out some possible solutions. 😃

**Reply by lol-powa** [reply]
Hi @genius3000 
Thanks a lot for your very helpful answers.
I'll try to get it working the way I want myself, if I can...
If not, I'll join the IRC support channel as you kindly suggested.
Have a nice day,
